include/osquery/http_client.h

```diff
@@ -0,0 +1,368 @@
+typedef beast_http::response_parser<beast_http::string_body>
+    beast_http_response_parser;
+typedef beast_http::request_serializer<beast_http::string_body>
+    beast_http_request_serializer;
+
+namespace osquery {
+namespace http {
+
+template <typename T>
+class HTTP_Request;
+template <typename T>
+class HTTP_Response;
+typedef HTTP_Request<beast_http_request> Request;
+typedef HTTP_Response<beast_http_response> Response;
+
+/**
+ * @brief http client class
+ *
+ * Implements put, post, get, head and delete_ methods.
+ * These methods take request as refrence and  return respose by value.
+ */
+class Client {
+ public: // struct(s)
+  class Options {
+   public:
+    Options()
+        : ssl_options_(0),
+          timeout_(0),
+          always_verify_peer_(false),
+          follow_redirects_(false) {}
+
+    Options& follow_redirects(bool fr) {
+      follow_redirects_ = fr;
+      return *this;
+    }
+
+    Options& always_verify_peer(bool avp) {
+      always_verify_peer_ = avp;
+      return *this;
+    }
+
+    Options& timeout(int to) {
+      timeout_ = to;
+      return *this;
+    }
+
+    Options& openssl_ciphers(std::string const& ciphers) {
+      ciphers_ = ciphers;
+      return *this;
+    }
+
+    Options& openssl_options(long so) {
+      ssl_options_ = so;
+      return *this;
+    }
+
+    Options& openssl_verify_path(std::string const& vp) {
+      verify_path_ = vp;
+      return *this;
+    }
+
+    Options& openssl_certificate(std::string const& scf) {
+      server_certificate_ = scf;
+      return *this;
+    }
+
+    Options& openssl_certificate_file(std::string const& ccf) {
+      client_certificate_file_ = ccf;
+      return *this;
+    }
+
+    Options& openssl_private_key_file(std::string const& cpkf) {
+      client_private_key_file_ = cpkf;
+      return *this;
+    }
+
+    Options& openssl_sni_hostname(std::string const& sni_h) {
+      sni_hostname_ = sni_h;
+      return *this;
+    }
+
+    Options& proxy_hostname(std::string const& prxy_h) {
+      proxy_hostname_ = prxy_h;
+      return *this;
+    }
+
+    Options& remote_hostname(std::string const& remote_h) {
+      remote_hostname_ = remote_h;
+      return *this;
+    }
+
+    Options& remote_port(std::string const& remote_p) {
+      remote_port_ = remote_p;
+      return *this;
+    }
+
+   private:
+    boost::optional<std::string> server_certificate_;
+    boost::optional<std::string> verify_path_;
+    boost::optional<std::string> client_certificate_file_;
+    boost::optional<std::string> client_private_key_file_;
+    boost::optional<std::string> ciphers_;
+    boost::optional<std::string> sni_hostname_;
+    boost::optional<std::string> proxy_hostname_;
+    boost::optional<std::string> remote_hostname_;
+    boost::optional<std::string> remote_port_;
+    long ssl_options_;
+    int timeout_;
+    bool always_verify_peer_;
+    bool follow_redirects_;
+    friend class Client;
+  };
+
+ public: // methods
```

Comment:
We don't need to name each, struct list/methods/etc.

Reply:
ok.